src/appMain/routes/project/routes/AssignGroup/ProjectRoleWidget/index.js

```diff
@@ -85,6 +87,22 @@ const ProjectRoleWidget = ({
           newChecked.splice(pai, 1);
         }
       }
+      if (value.metadata.name === "NAMESPACE_ADMIN") {
+        const pai = checked.findIndex(
+          (element) => element.metadata.name === "NAMESPACE_READ_ONLY"
+        );
+        if (pai !== -1) {
+          newChecked.splice(pai, 1);
+        }
+      }
+      if (value.metadata.name === "NAMESPACE_READ_ONLY") {
+        const pai = checked.findIndex(
+          (element) => element.metadata.name === "NAMESPACE_ADMIN"
+        );
+        if (pai !== -1) {
+          newChecked.splice(pai, 1);
+        }
+      }
```

Comment:
Can we simplify the whole things with something like:
```javascript
const pairs = ["PROJECT", "INFRA", "NAMESPACE"]
  .map((e) => [
    [e + "_ADMIN", e + "_READ_ONLY"],
    [e + "_READ_ONLY", e + "_ADMIN"],
  ])
  .reduce((a, b) => [...a, ...b], []);

for (let pair of pairs) {
  if (value.metadata.name === pair[0]) {
    const item = checked.findIndex(
      (element) => element.metadata.name === pair[1]
    );
    if (item !== -1) newChecked.splice(item, 1);
  }
}
```

Reply:
this is neat, have refactored all files applicable with above code.